File: sem3/map/Lab/3/toylanguage/src/main/java/com/company/Views/TextMenu.java
package com.company.Views;

import java.util.HashMap;
import java.util.Scanner;

public class TextMenu {
    private HashMap<String, Command> commands;
    private Scanner scanner;

    public TextMenu() {
        commands = new HashMap<>();
        this.scanner = new Scanner(System.in);
    }

    public void add(Command command) {
        this.commands.put(command.getName(), command);
    }

    private void printMenu() {
        System.out.println("\nWelcome!");
        for(Command command: this.commands.values()) {
            System.out.println(command);
        }
    }

    public void run() {
        while(true) {
            printMenu();
            System.out.println("Your command: ");
            String commandName = this.scanner.nextLine();

            Command command = this.commands.get(commandName);

            if(command != null) {
                command.execute();
            } else {
                System.out.println("Invalid command!");
            }
        }
    }
}

File: sem3/map/Lab/toylanguage_C#/ToyLanguage/ToyLanguage/Views/TextMenu.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Mime;

namespace ToyLanguage.Views
{
    public class TextMenu
    {
        private Dictionary<string, CommandBase> _commands;

        public TextMenu()
        {
            _commands = new Dictionary<string, CommandBase>();
        }

        public void AddCommand(CommandBase command)
        {
            _commands[command.Name] = command;
        }

        private void PrintMenu()
        {
            Console.WriteLine("\nWelcome!");
            _commands.ToList().ForEach(p => Console.WriteLine(p.Value));
        }

        public void Run()
        {
            while (true)
            {
                PrintMenu();
                Console.WriteLine("Your command: ");
                string line = Console.ReadLine();
                Console.ReadKey();

                CommandBase cmd = _commands.ContainsKey(line) ? _commands[line] : null;

                if (cmd != null)
                {
                    cmd.Execute();
                }
                else
                {
                    Console.WriteLine($"Invalid command + {line}!");
                }
            }
        }
        
    }
}